servers/rendering/renderer_viewport.cpp

```diff
@@ -286,11 +288,17 @@ void RendererViewport::_draw_viewport(Viewport *p_viewport) {
 				// The scene renderer will still copy over the last frame, so we need to clear the render target.
 				force_clear_render_target = true;
 			}
+
+			// Calculate the frames needed for volumetric fog to converge.
+			if (p_viewport->size != p_viewport->old_size && can_draw_3d) {
```

Comment:
Resizing the viewport is not the only thing that can require reconverging volumetric fog.

A lot of other things can require it for the volumetric fog preview to remain accurate:

- Toggling volumetric fog or changing one of its properties.
- Hiding or showing a FogVolume node, or changing one of its material properties.
- Hiding or showing a light that efects volumetric fog, or changing one of its properties.

This is a lot to keep track of, so it might be better to always redraw for the volumetric fog duration whenever volumetric fog is enabled. We should try to do this only if something 3D-related happened though, so that UI-only updates do not force a full redraw of both 2D and 3D (they are always drawn at the same time).